game-logic.js

```diff
@@ -0,0 +1,367 @@
+  generateOrderings(CURRENT_PLAYER, playerRoll); // record permutations
+
+  changeAction(ACTION_ORDER); // next action is to choose ordering
+
+  return (
+    displayCurrentPlayer(CURRENT_PLAYER) +
+    " rolled " +
+    playerRolls[CURRENT_PLAYER]
+  );
+};
+
+// user to supply index to the orderings of player roll
+var actionOrder = function (index) {
+  if (!isValidRollOrdering(CURRENT_PLAYER, index)) {
+    return `Invalid choice "${index}". Please re-enter`;
+  }
+  index = Number(index);
+  playerOrderingChoice[CURRENT_PLAYER] =
+    playersRollsOrderings[CURRENT_PLAYER][index]; // record the player choice of ordering
+
+  var desc = `${displayCurrentPlayer(
+    CURRENT_PLAYER
+  )} chose ${getDisplayOrdering(playerOrderingChoice[CURRENT_PLAYER])}. `;
+
+  if (CURRENT_PLAYER === LAST_PLAYER) {
+    // A round decision can be made. Decide the winner, reset game and change the next action to RESET
+    var playerWinners = decideRoundWinner();
+    console.log("winner" + playerWinners);
+    var currentDisplayWinners = playerWinners.map((playerWinner) =>
+      displayCurrentPlayer(playerWinner)
+    );
+
+    var resultDescription = `Winner: ${currentDisplayWinners}`;
+    PREV_ROUND_RESULT_DESCRIPTION = resultDescription;
+
+    var activityDescription = `Winner decided: ${currentDisplayWinners}`;
+
+    desc += activityDescription;
+    changeAction(ACTION_GAME_SET);
+  } else {
+    // A round decision cannot be made. Change the next action state to ROLL and player to next player
+    changeAction(ACTION_ROLL);
+    changePlayer();
+  }
+  return desc;
+};
+
+const actionGameSet = function () {
+  // game is set (winner decided) and button clicked
+  resetGame();
+  CURRENT_ACTION = ACTION_ROLL; // change the next action to ROLL.
+  return "";
+};
+/**
+ *  Main - On Button Click
+ *
+ *  Does some action based on current action state. actionXX(...) will execute
+ *  its logic and determine the next action state (remain or change, if any).
+ *
+ *  nextRenderParagraphXX(...) render output elements of paragraph to the webpage for the next action state (waiting for next click)
+ *
+ * */
+var main = function () {
+  var result = ``;
+  if (CURRENT_ACTION === ACTION_ROLL) {
+    result += `Previous activity: ${actionRoll()}`;
+  } else if (CURRENT_ACTION === ACTION_ORDER) {
+    var orderingChoiceCount = getOrderingChoiceCount(CURRENT_PLAYER);
+    var firstElementIndex = 0;
+    var value =
+      orderingChoiceCount === 1 ? firstElementIndex : HTML_G_INPUT_FIELD.value; // if there isn't a real choice, ignore user input and set value to first index
+    result += `Previous activity: ${actionOrder(value)}`;
+  } else if (CURRENT_ACTION === ACTION_GAME_SET) {
+    result += actionGameSet();
+  }
+  nextRenderParagraphMain(result);
+  nextRenderParagraphRound();
+};
+
+// nextRenderParagraphMain should be called AFTER the next action state has been updated.
+var nextRenderParagraphMain = function (result) {
```

Comment:
your DOM manipulation in this function is solid, but personal opinion its not the most readable. i understand your intentions to abstract away all the DOM manipulation by having the helper functions and the DOM elements stored in variables that are declared in another file. but when reading this I gotta constantly reference the other file to see exactly what element this is and then to the function definition to see exactly what it does, while taking into consideration the DOM elements it takes in